src/clients/c++/request.h

```diff
@@ -347,6 +364,23 @@ class InferContext {
     /// model configuration. Variable-size dimensions are reported as
     /// -1.
     virtual const DimsList& Dims() const = 0;
+
+    /// Set tensor values for this output by reference into a shared memory
+    /// region. The values are not copied and so it must not be modified or
+    /// destroyed until this output is ready (that is until after the Run()
+    //// call(s) have written the input completely). For batched outputs this
+    /// function must be called batch-size times to write all tensor values
+    /// for a batch of this output.
+    /// \param shm_key The identifier for the shared memory object that is used
+    /// to get the region in shared memory where the tensor values for this
+    /// output should be stored.
+    /// \param offset The offset into the shared memory to the start of the
+    /// output tensor values.
+    /// \param byte_size The size, in bytes of the output tensor data. Must
+    /// match the size expected by the output.
+    /// \return Error object indicating success or failure.
+    virtual Error SetSharedMemory(
```

Comment:
Delete? We don't set output values...

Reply:
This just sets the shared memory region to be used for the output tensors. it does not set their values

Reply:
Right. The comment needs to be changed to indicate that this is just providing shared memory buffer where output is to be written... as it is now it mentions output values.